frontend/src/scenes/session-recordings/player/metaLogic.ts

```diff
@@ -37,33 +47,26 @@ export const metaLogic = kea<metaLogicType>({
                 return [device, location].filter((s) => s).join(' · ')
             },
         ],
-        currentSnapshot: [
-            (selectors) => [selectors.snapshots, selectors.time],
-            (snapshots, time): eventWithTime | null => {
-                const index = snapshots.findIndex((s: eventWithTime) => s.timestamp > time.current && 'width' in s.data)
-                if (index === -1) {
-                    return null
-                }
-                return snapshots[index]
-            },
-        ],
         resolution: [
-            (selectors) => [selectors.currentSnapshot],
-            (snapshot) => {
-                if (!(snapshot && 'width' in snapshot.data && 'height' in snapshot.data)) {
-                    return null
+            (selectors) => [selectors.snapshots, selectors.time],
+            (snapshots, time) => {
+                // Find snapshot to pull resolution from
+                const lastIndex = findLastIndex(snapshots, (s: eventWithTime) => 'width' in s.data)
+                if (lastIndex === -1) {
+                    return {
```

Comment:
Nit: Maybe we should default to null here and leave the field blank in that case. Otherwise we risk showing the wrong information.

Reply:
good idea!